#!/usr/bin/env python
PACKAGE = "road_marking_localization"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("blur_kernel_size", int_t, 0, "An Integer parameter", 1, 1, 20)
gen.add("crop_top_pixels", int_t, 0, "An Integer parameter", 50, 0, 640)
gen.add("x_box", double_t, 0, "An Integer parameter", 2.0, -10.0, 10.0)
gen.add("y_box", double_t, 0, "An Integer parameter", 2.0, -10.0, 10.0)
gen.add("minimum_z", double_t, 0, "An Integer parameter", -0.02, -1.0, 1.0)
gen.add("maximum_z", double_t, 0, "An Integer parameter", 0.02, -1.0, 1.0)
gen.add("threshold", int_t, 0, "An Integer parameter", 160, 0, 255)
gen.add("icp_max_iterations", int_t, 0, "An Integer parameter", 5, 0, 100)
gen.add("icp_RANSAC_outlier_rejection_threshold", double_t, 0, "An Integer parameter", 0.0, 0.0, 1.0)
gen.add("icp_RANSAC_iterations", int_t, 0, "An Integer parameter", 0, 0, 1000)
gen.add("icp_max_correspondence_distance", double_t, 0, "An Integer parameter", 0.10, 0.0, 20.0)
gen.add("icp_sample_size", int_t, 0, "An Integer parameter", 750, 0, 5000)
gen.add("minimum_points", int_t, 0, "An Integer parameter", 250, 0, 5000)
gen.add("maximum_x_correction", double_t, 0, "An Integer parameter", 0.3, 0.0, 1.0)
gen.add("maximum_y_correction", double_t, 0, "An Integer parameter", 0.3, 0.0, 1.0)
gen.add("maximum_yaw_correction", double_t, 0, "An Integer parameter", 0.5, -3.141, 3.141)
gen.add("debug", bool_t, 0, "An Integer parameter", True)

transformation_estimations = gen.enum([ gen.const("levenberg_marquardt",      int_t, 0, "A small constant"),
                             gen.const("svd",     int_t, 1, "A medium constant"),
                             gen.const("svd2D",     int_t, 2, "A medium constant"),
                             gen.const("center_of_mass",      int_t, 3, "A large constant"),
                             gen.const("dual_quaternion",      int_t, 4, "A large constant")],
                             "Transformation estimation type")

gen.add("transformation_estimation", int_t, 0, "Transformation estimation type", 0, 0, 4, edit_method=transformation_estimations)

exit(gen.generate(PACKAGE, "road_marking_localization", "RoadMarkingLocalization"))
